quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/services/metrics/events/MicrometerMetricsEventListener.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.quarkus.runtime.services.metrics.events;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Metrics;
+import org.jboss.logging.Logger;
+import org.keycloak.events.Details;
+import org.keycloak.events.Errors;
+import org.keycloak.events.Event;
+import org.keycloak.events.EventListenerProvider;
+import org.keycloak.events.EventListenerTransaction;
+import org.keycloak.events.EventType;
+import org.keycloak.events.admin.AdminEvent;
+import org.keycloak.events.admin.OperationType;
+import org.keycloak.models.KeycloakSession;
+
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class MicrometerMetricsEventListener implements EventListenerProvider {
+
+    private static final Logger logger = Logger.getLogger(MicrometerMetricsEventListener.class);
+
+    private static final EnumSet<EventType> LOGIN_EVENT_TYPES =
+            EnumSet.of(EventType.LOGIN, EventType.LOGIN_ERROR);
+
+    private static final String PROVIDER_KEYCLOAK_OPENID = "keycloak";
+    private static final String REALM_TAG = "realm";
+    private static final String PROVIDER_TAG = "provider";
+    private static final String CLIENT_ID_TAG = "client.id";
+    private static final String ERROR_TAG = "error";
+    private static final String RESOURCE_TAG = "resource";
+
+    private static final String KEYLOAK_METER_NAME_PREFIX = "keycloak.";
+    private static final String EVENT_PREFIX = KEYLOAK_METER_NAME_PREFIX + "event.";
+    private static final String LOGIN_COUNTER_NAME = EVENT_PREFIX + "login";
+    private static final String LOGIN_ATTEMPT_COUNTER_NAME = EVENT_PREFIX + "login.attempt";
+    private static final String LOGIN_ERROR_COUNTER_NAME = EVENT_PREFIX + "login.error";
+    private static final String ADMIN_EVENT_PREFIX = KEYLOAK_METER_NAME_PREFIX + "admin.event.";
+
+    static final Map<OperationType, String> ADMIN_OPERATION_TYPE_TO_NAME =
```

Comment:
Did we need a counter for each OperationType? I guess not, an operation tag might be better